packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -603,6 +603,11 @@ export function cache(
         args.unshift(boundArgs)
       }
 
+      // When draft mode is enabled, we must bypass caching altogether.
+      if (workStore.isDraftMode) {
+        return fn.apply(null, args)
```

Comment:
I think we can land this now but this does open up the possibility of slightly different runtime semantics. For instance if we serialize iterables as arrays then most of the time on the inside the runtime value of an iterable will be an array. We're already in dangerous territory here b/c TS is not aware of this downcasting of inputs across the boundary. But once it is then we'll have a different problem where TS thinks it is the downcast type but in draft mode only it's the original type.

Maybe we leave a TODO comment here to revisit later. Or you can just update it to maintain the serialization boundary

Reply:
That's a good point. Not only should we serialize/deserialize the args, but then we must also process the return value as we would when Draft Mode is disabled, otherwise temporary references wouldn't be resolved, for example. Furthermore, logs should also be associated with the Cache environment if Draft Mode is enabled. I've changed the implementation to only bypass cache reads and writes, everything else is handled the same as when Draft Mode is disabled.